### PR TITLE
adds alternative remotes blog

### PR DESCRIPTION
**Pull request by tomjemmett**

### CODE REVIEW

File: blogs/posts/2023-04-26_alternative_remotes.qmd [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+All of the steps below are intended to be entered via a terminal.
+
+Let's imagine that we have a GitHub account called example and a repository called test, and we use https rather than ssh.
+
+```
+$ git remote get-url origin
+# https://github.com/example/test.git
+```
+
+Now, let's say we have someone who has submitted a PR, and their username is friend. We can add a new remote for their fork with
```
Comment (Lextuga007) [inline blogs/posts/2023-04-26_alternative_remotes.qmd:26]: Can we write pull request (PR)? It's part of the style for writing that I'd like to introduce https://nhs-r-community.github.io/NHSR-way/community-handbook.html#explain-acronyms from NHS-R

**Review by Lextuga007 — CHANGES_REQUESTED** [review]

File: blogs/posts/2023-04-26_alternative_remotes.qmd [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+---
+
+It's great when someone send's you a pull request on GitHub to fix bugs or add new features to your project, but you probably always want to check the other persons work in someway before merging that pull request.
+
+All of the steps below are intended to be entered via a terminal.
+
+Let's imagine that we have a GitHub account called example and a repository called test, and we use https rather than ssh.
+
+```
+$ git remote get-url origin
```
Comment (Lextuga007) [inline blogs/posts/2023-04-26_alternative_remotes.qmd:22]: I stupidly copied this from GitHub's clone/fork button thing and then tried to type `git remote get-https://github.com/The-Strategy-Unit/data_science.git origin` which of course didn't work!

File: blogs/posts/2023-04-26_alternative_remotes.qmd [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+# https://github.com/example/test.git
+```
+
+Now, let's say we have someone who has submitted a PR, and their username is friend. We can add a new remote for their fork with
+
+```
+$ git remote add-url friend https://github.com/friend/test.git
```
Comment (Lextuga007) [inline blogs/posts/2023-04-26_alternative_remotes.qmd:29]: Should this be `git remote add friend https://github.com/friend/test.git` and omit the -url?

File: blogs/posts/2023-04-26_alternative_remotes.qmd [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+knitr:
+  opts_chunk:
+    fig.height: 2.5
+    echo: false
+    dev: png
+    dev.args:
+      bg: transparent
+---
+
+It's great when someone send's you a pull request on GitHub to fix bugs or add new features to your project, but you probably always want to check the other persons work in someway before merging that pull request.
```
[the remark this answers is not in the document]
Comment (tomjemmett) [reply blogs/posts/2023-04-26_alternative_remotes.qmd:15]: ah, that's reminded me I need to reinstall grammarly plugin!

[the remark this answers is not in the document]
Comment (Lextuga007) [reply blogs/posts/2023-04-26_alternative_remotes.qmd:15]: Is that in VS Code?

[the remark this answers is not in the document]
Comment (tomjemmett) [reply blogs/posts/2023-04-26_alternative_remotes.qmd:15]: yes, there is a plugin that works great in VS Code. Though, this [rstudio issue](https://github.com/rstudio/rstudio/issues/12462) suggests that maybe there is some support in RStudio.

File: blogs/posts/2023-04-26_alternative_remotes.qmd (hunk 3 above)
Comment (tomjemmett) [reply blogs/posts/2023-04-26_alternative_remotes.qmd:29]: yes, it absolutely should!

File: blogs/posts/2023-04-26_alternative_remotes.qmd (hunk 1 above)
Comment (tomjemmett) [reply blogs/posts/2023-04-26_alternative_remotes.qmd:26]: changed!

File: blogs/posts/2023-04-26_alternative_remotes.qmd [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+
+```
+# make sure to replace the remote name to what you set it to before
+$ git fetch friend
+```
+
+Now, hopefully they have commited to a branch with a name that you haven't used. Let's say they created a branch called `my_work`. You can then simply run
+
+```
+$ git switch my_work
```
[the remark this answers is not in the document]
Comment (tomjemmett) [reply blogs/posts/2023-04-26_alternative_remotes.qmd:44]: I think you are right. I wrote this just from memory as I didn't have an obvious test repository to play with!

File: blogs/posts/2023-04-26_alternative_remotes.qmd [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+Now, if they have happened to use a branch name that you are already using, or more likely, directly commited to their own `main` branch, you will need to do checkout to a new branch:
+
+```
+# replace friend as above to be the name of the remote, and main to be the branch
+# that they have used
+# replace their_work with whatever you want to call this branch locally
+$ git checkout friend/main -b their_work
+```
+
+Finally, you are now ready to run their code and check everything is good to merge!
```
[the remark this answers is not in the document]
Comment (tomjemmett) [reply blogs/posts/2023-04-26_alternative_remotes.qmd:58]: in this example you shouldn't end up in a detatched branch state. If you do, then to clean up you just need to checkout an existing branch, no clean up needed.

The clean up that you might want to do would be to delete the branch you checked out, and then remove the new remote.

**Review by Lextuga007 — APPROVED** [review]

**Pull request merged**